misc/ls-extensions/modules/performance-analyzer-services/src/main/java/io/ballerina/ProgramAnalyzerNodeVisitor.java

```diff
@@ -375,1133 +196,165 @@ public void visit(FunctionDefinitionNode functionDefinitionNode) {
+                    long lhsValue = Long.parseLong(((BinaryExpressionNode) node).lhsExpr().toString().trim());
+                    iterationsCount = rhsValue - lhsValue + 1;
+                }
+            }
+        }
 
-    @Override
-    public void visit(ClassDefinitionNode classDefinitionNode) {
+        ForStatementNode forStatementNode = new ForStatementNode(iterationsCount);
+        Node currentParentNode = this.currentNode;
 
-    }
+        Node forBodyNode = new Node();
+        this.currentNode = forBodyNode;
 
-    @Override
-    public void visit(ResourcePathParameterNode resourcePathParameterNode) {
+        visitSyntaxNode(forEachStatementNode.blockStatement());
+        forStatementNode.setForBody(forBodyNode.getNextNode());
 
-    }
+        this.currentNode = currentParentNode;
 
-    @Override
-    public void visit(RequiredExpressionNode requiredExpressionNode) {
+        if (forEachStatementNode.blockStatement() != null) {
+            this.setChildNode(forStatementNode);
+        }
 
+        visitSyntaxNode(forEachStatementNode);
     }
 
     @Override
-    public void visit(ErrorConstructorExpressionNode errorConstructorExpressionNode) {
-
-    }
+    public void visit(WhileStatementNode whileStatementNode) {
 
-    @Override
-    public void visit(ParameterizedTypeDescriptorNode parameterizedTypeDescriptorNode) {
+        ForStatementNode forStatementNode = new ForStatementNode(DEFAULT_LOOP_SIZE);
+        Node currentParentNode = this.currentNode;
 
-    }
+        Node forBodyNode = new Node();
+        this.currentNode = forBodyNode;
 
-    @Override
-    protected void visitSyntaxNode(io.ballerina.compiler.syntax.tree.Node node) {
+        visitSyntaxNode(whileStatementNode.whileBody());
+        forStatementNode.setForBody(forBodyNode.getNextNode());
 
-        super.visitSyntaxNode(node);
+        this.currentNode = currentParentNode;
+        if (forStatementNode.getForBody() != null) {
+            this.setChildNode(forStatementNode);
+        }
     }
 
     @Override
-    public int hashCode() {
+    public void visit(RemoteMethodCallActionNode remoteMethodCallActionNode) {
 
-        return super.hashCode();
+        resolveActionInvocation(remoteMethodCallActionNode);
+        visitSyntaxNode(remoteMethodCallActionNode);
     }
 
     @Override
-    public boolean equals(Object obj) {
+    public void visit(AssignmentStatementNode assignmentStatementNode) {
 
-        return super.equals(obj);
-    }
+        if (assignmentStatementNode.varRef().kind() == SyntaxKind.SIMPLE_NAME_REFERENCE &&
+                assignmentStatementNode.expression().kind() == SyntaxKind.SIMPLE_NAME_REFERENCE) {
 
-    @Override
-    protected Object clone() throws CloneNotSupportedException {
+            Optional<Symbol> referenceSymbol = model.symbol(assignmentStatementNode.varRef());
 
-        return super.clone();
-    }
+            if (referenceSymbol.isEmpty()) {
```

Comment:
We can merge the checks for reference and expression symbols together

Reply:
Fixed all.